install.ts

```diff
@@ -0,0 +1,202 @@
+
+function panic(err: Error): void {
+  if (err) console.error("[deno-install error]", err.stack);
+  console.error("[deno-install error]", "Installation failed");
+  exit(1);
+}
+
+function pinup(...args: any): void {
+  console.log("[deno-install info]", ...args);
+}
+
+async function follow(url: string): Promise<any> {
+  let located: boolean = false;
+  let res: any; // TODO: annotate deno Response
+  let count: number = 0;
+  while (!located) {
+    if (++count === 4) throw Error(`Unable to fetch from ${url}`);
+    res = await fetch(url);
+    if (res.status >= 300 && res.status < 400)
+      url = res.headers.get("Location");
+    else if (res.status === 200) located = true;
+  }
+  return res;
+}
+
+async function releaseUrl(tag?: string): Promise<{ [key: string]: string }> {
+  const url: string = tag ? `${TAG_RELEASE_URL}/${tag}` : LATEST_RELEASE_URL;
+  let filename: string;
+  switch (platform.os) {
+    case "linux":
+      filename = LINUX_GZIP;
+      break;
+    case "mac":
+      filename = OSX_GZIP;
+      break;
+    case "win":
+      filename = WIN_ZIP;
+      break;
+    default:
+      throw Error(`Unsupported operating system ${platform.os}`);
+  }
+  const res: any = await follow(url); // TODO: annotate deno Response
+  const link: string = (await res.text())
+    .split(/\r?\n/)
+    .find((line: string) => line.includes("href") && line.includes(filename));
+  if (!link) panic(Error(`Unable to find ${filename} @ ${url}`));
+  const match: string = link.replace(/^.*href=(?:"|')([^"']*)(?:"|').*$/, "$1");
+  if (!/^\/denoland/.test(match))
+    panic(Error(`Unable to find ${filename} @ ${url}`));
+  return {
+    url: `https://github.com${match}`,
+    tag: match.replace(/^.*(v\d+\.\d+\.\d+).*$/, "$1")
+  };
+}
+
+async function tempDownload(
+  tempDir: string,
+  url: string,
+  suffix: string
+): Promise<string> {
+  const res: any = await follow(url); // TODO: annotate deno Response
+  const tempFile: string = join(tempDir, `${Date.now()}.${suffix}`);
+  await writeFile(tempFile, new Uint8Array(await res.arrayBuffer()));
+  return tempFile;
+}
+
+async function unpackBin(archive: string): Promise<void> {
+  await mkdirp(DENO_BIN_DIR);
+  let args: string[];
+  if (WIN32) {
+    await rename(DENO_BIN, OLD_DENO_BIN);
+    args = [
+      "powershell.exe",
+      "-Command",
+      `Expand-Archive "${archive}" -DestinationPath "${DENO_BIN_DIR}"`
+    ];
+  } else {
+    args = ["gunzip", "-d", archive];
+  }
+  const child: Process = run({ args });
+  const childStatus: ProcessStatus = await child.status();
+  if (!childStatus.success)
+    panic(Error(`(g)unzip failed. ${args} -> ${childStatus.code}`));
+  child.close();
+  if (!WIN32) {
+    if (platform.os === "linux") await rename(DENO_BIN, OLD_DENO_BIN);
+    await copyFile(archive.replace(/\.gz$/, ""), DENO_BIN);
+  }
+}
+
+async function makeHandy(): Promise<void> {
+  if (WIN32) {
+    if (!procEnv.Path.includes(DENO_BIN_DIR)) {
```

Comment:
Windows file paths are case-insensitive, `String#includes` is not.

```suggestion
    if (!procEnv.Path.toLocaleLowerCase().includes(DENO_BIN_DIR.toLocaleLowerCase())) {
```